relayer/relays/beacon/header/syncer/syncer.go

```diff
@@ -133,7 +154,16 @@ func (s *Syncer) GetSyncCommitteePeriodUpdate(from uint64) (scale.Update, error)
 
 	blockRootsProof, err := s.GetBlockRoots(uint64(finalizedHeader.Slot))
 	if err != nil {
-		return scale.Update{}, fmt.Errorf("fetch block roots: %w", err)
+		beaconStateData, err := s.store.GetBeaconStateData(uint64(finalizedHeader.Slot))
+		if err != nil {
+			return scale.Update{}, fmt.Errorf("fetch beacon state for block roots proof: %w", err)
+		}
+		beaconState, err := s.unmarshalBeaconState(uint64(finalizedHeader.Slot), beaconStateData)
+
+		blockRootsProof, err = s.GetBlockRootsFromState(beaconState)
```

Comment:
If the state is not available, check if it perhaps available in the beacon store.